src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -165,17 +171,13 @@ public override BoundNode VisitITuplePattern(BoundITuplePattern node)
                         // for property part
                         if (!rp.Properties.IsDefault)
                         {
-                            foreach (BoundSubpattern subpattern in rp.Properties)
+                            foreach (BoundPropertySubpattern subpattern in rp.Properties)
                             {
-                                if (subpattern.Symbols.IsEmpty)
-                                    continue;
-
-                                // Obtain the last symbol's slot which is actually the one being matched.
-                                Symbol last = subpattern.Symbols.Last();
-                                // PROTOTYPE(extended-property-patterns) For a chain of members, the input type is no longer `inputType`
-                                if (last.ContainingType.Equals(inputType, TypeCompareKind.AllIgnoreOptions))
+                                // PROTOTYPE(extended-property-patterns): Investigate if we need to visit nested members; is there a test gap?
+                                if (subpattern.Member is { Symbol: Symbol symbol, Receiver: var receiver } member &&
+                                    symbol.ContainingType.Equals(receiver is not null ? receiver.Type : inputType, TypeCompareKind.AllIgnoreOptions))
```

Comment:
>symbol.ContainingType.Equals(receiver is not null ? receiver.Type : inputType, TypeCompareKind.AllIgnoreOptions) [](http://example.com/codeflow?start=36&length=112)

It's not clear to me why we compare the containing type to the receiver or input type using `Equals()`. What if the receiver or input type is a derived type?

Reply:
The logic is unchanged from before but it sounds like the check is actually redundant. Remove?

Reply:
I do recall one or two tests failing without the check but I couldn't find it now. I'll remove it to see if CI fails.

Reply:
If no tests fail without this check, is there a simple test with a derived type that we could add that would demonstrate the difference in behavior?

Reply:
I *think* this is the one: RecursivePatternNullInferenceWithDowncast_01 but now it's green.

Reply:
I actual went back to main and rechecked. No test fails. I might be missing something though.

Reply:
This probably is where it's fixed: https://github.com/dotnet/roslyn/pull/51950